### PR TITLE
Add xeno-canto urls to species generator; closes #12

### LINKED ISSUE #12: Add the bird sound urls to each species

### PR DESCRIPTION
**Pull request by EmilyMB**
This pull request adds additional logic to the SpeciesGenerator to add species to the database from the [Xeno Canto API](http://www.xeno-canto.org/article/153).

Changes include:
- Add species from Xeno Canto to the database.
- Add sound clip url(when available) to each species from the [GBIF API](http://gbif.blogspot.com/).
- Fix gbif_service_test.rb based on changes to the API processing.

### CODE REVIEW

File: app/services/xeno_canto_service.rb [hunk 1]
```diff
@@ -11,9 +11,23 @@ def recordings(min_lat, min_long, max_lat, max_long)
       .get("recordings?query=box:#{min_lat},#{min_long},#{max_lat},#{max_long}"))
   end
 
+  def species_recording(species)
+    species_parse(connection
+      .get("recordings?query=#{species}"))
```
Comment (EmilyMB) [inline app/services/xeno_canto_service.rb:16]: New Rubies allow this, and I think it improves clarity.

Comment (cluhring) [inline app/services/xeno_canto_service.rb:16]: Thats brash to claim you are smarter than the hound.

File: test/services/xeno_canto_service_test.rb [hunk 2]
```diff
@@ -24,4 +24,18 @@ def setup
...
+      recording = service.species_recording("troglodytes troglodytes")
+      assert_equal "http://www.xeno-canto.org/233278/download", recording
+    end
+  end
+
+  test "recording by species when not found" do
+    VCR.use_cassette("xeno_canto_service_fake_species_recordings") do
+      recording = service.species_recording("fake bird")
+      assert_equal "not found", recording
+    end
```
Comment (cluhring) [inline test/services/xeno_canto_service_test.rb:39]: nice tests Emily. Get a :haircut:, hippy

File: app/generator/api_combiner.rb [hunk 3]
```diff
@@ -1,27 +1,57 @@
 module ApiCombiner
+  attr_reader :trail_species
+
   def self.get_species(trail_id)
-    gbif = GbifService.new
-    eol = EolService.new
     trail = Trail.find(trail_id)
-    trail_species = []
+    @trail_species = []
```
Comment (cluhring) [inline app/generator/api_combiner.rb]: Doesn't really look like you are doing anything with this instance variable & attr_reader, necessary?

File: app/generator/api_combiner.rb [hunk 4]
```diff
@@ -1,27 +1,57 @@
...
-          clip_url: "not available",
+          clip_url: xeno_canto.species_recording(species["species"]),
           lat: species["decimalLatitude"],
           lng: species["decimalLongitude"])
       end
     end
-    trail_species
+  end
+
+  def self.add_xeno_canto_species(trail_id, lat, lng)
```
Comment (Lydias303) [inline app/generator/api_combiner.rb:32]: I think this is fine.

File: app/services/xeno_canto_service.rb (hunk 1 above)
Comment (Lydias303) [inline app/services/xeno_canto_service.rb:16]: "We" made a decision Chris.

File: test/services/xeno_canto_service_test.rb (hunk 2 above)
Comment (Lydias303) [inline test/services/xeno_canto_service_test.rb:39]: Good testing!

**Pull request merged**